src/objects.c

```diff
@@ -2175,6 +2175,65 @@ int p11prov_obj_get_ec_public_x_y(P11PROV_OBJ *obj, CK_ATTRIBUTE **pub_x,
     return ret;
 }
 
+CK_ATTRIBUTE *p11prov_obj_get_ec_public_raw(P11PROV_OBJ *key)
+{
+    CK_ATTRIBUTE *pub_key;
+
+    if (!key) {
+        return RET_OSSL_ERR;
+    }
+
+    if (key->data.key.type != CKK_EC) {
+        P11PROV_raise(key->ctx, CKR_GENERAL_ERROR, "Unsupported key type");
+        return RET_OSSL_ERR;
+    }
+
+    if (key->class != CKO_PRIVATE_KEY && key->class != CKO_PUBLIC_KEY) {
+        P11PROV_raise(key->ctx, CKR_GENERAL_ERROR, "Invalid Object Class");
+        return RET_OSSL_ERR;
+    }
+
+    pub_key = p11prov_obj_get_attr(key, CKA_P11PROV_PUB_KEY);
+    if (!pub_key) {
+        CK_ATTRIBUTE *ec_point;
+
+        ec_point = p11prov_obj_get_attr(key, CKA_EC_POINT);
+        if (ec_point) {
```

Comment:
Would not be nicer to reverse this condition to avoid nesting the whole processing below?

Reply:
Maybe, but given this works and it is not terribly ugly, I'll take your LGTM and just merge as is :)